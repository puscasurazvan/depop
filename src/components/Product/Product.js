import React, { useState } from 'react'
import PropTypes from 'prop-types'

import like from '../../assets/like.svg'

import './Product.scss'

const Product = ({
  productTitle,
  productBrand,
  productSize,
  productPrice,
  imageURL,
  sold,
  handleLike,
  addProductLikes,
}) => {
  const [liked, setLiked] = useState(false)

  const handleLiked = productTitle => {
    setLiked(!liked)
    addProductLikes(productTitle)
  }

  return (
    <div className="productWrapper">
      <div className="imagewrapper">
        <img
          src={imageURL}
          alt="product display"
          className={sold ? 'imagewrapper__sold' : 'imagewrapper__image'}
          lazy="loading"
        />
        {sold && <p className="imagewrapper__soldText">SOLD</p>}
      </div>
      <p className="title"> {productTitle}</p>
      <p className="brand"> {productBrand}</p>
      <p className="size"> {productSize}</p>
      <p className="price"> Â£{productPrice}</p>
      <button
        onClick={() => handleLiked(productTitle)}
        className={liked ? 'button liked' : 'button'}
      >
        <img src={like} className="like" alt="" />
      </button>
    </div>
  )
}

Product.defaultProps = {
  productTitle: 'No description specified',
  productBrand: 'No brand specified',
  productSize: 'No size specified',
  productPrice: 12,
}

Product.propTypes = {
  imageURL: PropTypes.string.isRequired,
  productTitle: PropTypes.string.isRequired,
  productBrand: PropTypes.string.isRequired,
  productSize: PropTypes.string.isRequired,
  productPrice: PropTypes.number.isRequired,
}

export default Product
