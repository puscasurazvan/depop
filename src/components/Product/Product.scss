.productWrapper {
  margin: 25px;
  position: relative;
  width: 220px;
  .imagewrapper {
    &__image {
      width: 220px;
      height: 150px;
      border-radius: 5px;
      position: relative;
      position: relative;
    }
    &__sold {
      width: 220px;
      height: 150px;
      border-radius: 5px;
      opacity: 0.35;
    }
    &__soldText {
      color: white;
      text-shadow: 1px 1px rgb(155, 155, 155);
      font-weight: bold;
      font-size: 18px;
      position: absolute;
      top: 20%;
      left: 35%;
    }
  }
  .title,
  .brand,
  .size,
  .price {
    margin: 0;
    font-weight: bold;
    font-size: 18px;
  }
  .button {
    position: absolute;
    top: 0;
    right: 0;
    background: transparent;
    background-color: white;
    border: 1px solid rgb(136, 136, 136);
    margin: 5px;
    border-radius: 5px;
    transition: 0.3s;
    opacity: 0.7;
    .like {
      width: 25px;
      height: 25px;
      margin: 0;
      padding: 0;
    }
  }
  .button:hover {
    opacity: 1;
  }
  .liked {
    opacity: 0.9;
    background-color: red;
    transition: 0.2s;
  }
}
