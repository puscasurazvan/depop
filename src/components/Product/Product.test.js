import React from 'react'
import { mount } from 'enzyme'
import Product from './Product'

const minProps = {
  productTitle: 'No description specified',
  productBrand: 'No brand specified',
  productSize: 'No size specified',
  productPrice: 200,
  imageURL: 'https://image.url',
}

describe('Product', () => {
  it('should render with the right props', () => {
    const wrapper = mount(<Product {...minProps} />)
    expect(wrapper.props().productBrand).toBe('No brand specified')
    expect(wrapper.props().productTitle).toBe('No description specified')
    expect(wrapper.props().productSize).toBe('No size specified')
    expect(wrapper.props().productPrice).toBe(200)
    expect(wrapper.find('Product').length).toEqual(1)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<Product {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
