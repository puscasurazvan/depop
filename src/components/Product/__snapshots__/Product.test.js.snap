// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render without exploding 1`] = `ReactWrapper {}`;
